### PR TITLE
Fix storage of stack trace of exception from reflection

### PR DESCRIPTION
**Pull request by janvorli**
There was one more case where we have saved the stack trace into the `_remoteStackTraceString` field in the exception when the exception was passing from reflection invoked code to the caller of that code. While there is no visible difference in the `Exception.ToString`, a SOS test was failing due to that. And it is not necessary to save the stack trace there.

I have thought about the cases when we really need the stack trace saved into the `_remoteStackTraceString` and I believe that actually the only case is when an existing exception is thrown again in managed code. So I have removed the option to save the stack trace from all the variants of the `DispatchManagedException`.

### CODE REVIEW

**Comment by janvorli** [comment]
cc: @mikem8361

**Review by mikem8361 — APPROVED** [review]

File: src/coreclr/vm/exceptionhandling.cpp [hunk 1]
```diff
@@ -5661,19 +5661,12 @@ VOID DECLSPEC_NORETURN DispatchManagedException(OBJECTREF throwable, CONTEXT* pE
 
     Thread *pThread = GetThread();
 
-    if (preserveStackTrace)
-    {
-        pThread->IncPreventAbort();
-        ExceptionPreserveStackTrace(throwable);
```
Comment (jkotas) [inline src/coreclr/vm/exceptionhandling.cpp:5667]: I am wondering whether the one remaining use of `ExceptionPreserveStackTrace` in the VM is actually necessary. It is wrapped in very confusing logic that does not look right and we seem to be getting lucky that it never kicks in practice.

It is fine to do clean it up in a separate PR.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/vm/exceptionhandling.cpp (hunk 1 above)
Comment (janvorli) [reply src/coreclr/vm/exceptionhandling.cpp:5667]: Thank you, I'll look into it separately.

**Comment by janvorli** [comment]
/ba-g the changes are in coreclr, the legs that are failing (timing out) are NativeAOT builds only.

**Comment by mikem8361** [comment]
Are we going to backport this to 9.0?

**Comment by janvorli** [comment]
@mikem8361 yes, I am planning to do that.
